scanner/src/main/kotlin/scanners/fossid/Utils.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners.fossid
+
+import org.ossreviewtoolkit.clients.fossid.model.rules.IgnoreRule
+import org.ossreviewtoolkit.clients.fossid.model.rules.RuleType
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.PathExclude
+
+private val DIRECTORY_REGEX = "(?<directory>(?:\\w+/)*\\w+)/?(?<starstar>\\*\\*)?".toRegex()
+private val EXTENSION_REGEX = "\\*\\.(?<extension>\\w+)".toRegex()
+private val FILE_REGEX = "(?<file>[\\w.]+)".toRegex()
+
+/**
+ * Convert an ORT path excludes in [excludes] to FossID [IgnoreRule]. If an error is encountered during the mapping, an
+ * issue is added to [issues].
+ *
+ */
+fun convertRules(excludes: Excludes, issues: MutableList<Issue>): List<IgnoreRule> {
+    return excludes.paths.mapNotNull {
+        it.mapToRule().also { mappedRule ->
+            if (mappedRule == null) {
+                issues += Issue(
+                    source = "FossID.convertRules",
+                    message = "Path exclude '${it.pattern}' cannot be converted to an ignore rule.",
+                    severity = Severity.HINT
+                )
+                FossId.logger.warn {
+                    "Path exclude  '${it.pattern}' cannot be converted to an ignore rule."
+                }
+            }
+        }
+    }
+}
+
+private fun PathExclude.mapToRule(): IgnoreRule? {
+    val directoryMatch = DIRECTORY_REGEX.matchEntire(pattern)
+    directoryMatch?.let {
+        val directory = it.groups["directory"]!!.value
+        val starStar = it.groups["starstar"]?.value
```

Comment:
Could now be written as

    DIRECTORY_REGEX.matchEntire(pattern)?.let { directoryMatch ->
        val directory = directoryMatch.groups["directory"]!!.value
        val starStar = directoryMatch.groups["starstar"]?.value

Similar below.